presto-main/src/main/java/com/facebook/presto/execution/ClusterSizeMonitor.java

```diff
@@ -224,6 +254,36 @@ public synchronized ListenableFuture<?> waitForMinimumCoordinators()
         return future;
     }
 
+    public synchronized ListenableFuture<?> waitForMinimumCoordinatorSidecars()
+    {
+        if (currentCoordinatorSidecarCount == 1 || !isCoordinatorSidecarEnabled) {
+            return immediateFuture(null);
+        }
+
+        SettableFuture<?> future = SettableFuture.create();
+        coordinatorSidecarSizeFutures.add(future);
+
+        // if future does not finish in wait period, complete with an exception
+        ScheduledFuture<?> timeoutTask = executor.schedule(
+                () -> {
+                    synchronized (this) {
+                        future.setException(new PrestoException(
+                                GENERIC_INSUFFICIENT_RESOURCES,
+                                format("Insufficient active coordinator sidecar nodes. Waited %s for at least 1 coordinator sidecars, but only %s coordinator sidecars are active", coordinatorSidecarMaxWait, currentCoordinatorSidecarCount)));
```

Comment:
It should be 0 or 1, so we don't need to parameterize `currentCoordinatorSidecarCount`